src/gf-platformid

```diff
@@ -41,14 +41,35 @@ coreos_gf_run_mount ro "${tmp_dest}"
 # We just mount the boot partition writable
 coreos_gf remount /boot rw:true
 
+# Look up platform-specific configuration
+# Legacy defaults if platforms.json is missing
+extra_grub_cmds="serial --speed=115200\nterminal_input serial console\nterminal_output serial console"
+extra_kargs="console=tty0 console=${DEFAULT_TERMINAL},115200n8"
+if [ "$(coreos_gf exists /boot/coreos/platforms.json)" = "true" ]; then
+    coreos_gf download /boot/coreos/platforms.json "${tmpd}"/platforms.json
+    # If the platform isn't in platforms.json, set both vars to ""
+    extra_grub_cmds=$(jq -r ".${platformid}.grub_commands // [] | join(\"\\n\")" < "${tmpd}/platforms.json")
+    extra_kargs=$(jq -r ".${platformid}.kernel_arguments // [] | join(\" \")" < "${tmpd}/platforms.json")
+fi
+
 # Inject PLATFORM label in BLS config (for subsequent config regeneration)
 blscfg_path=$(coreos_gf glob-expand /boot/loader/entries/ostree-*.conf)
 coreos_gf download "${blscfg_path}" "${tmpd}"/bls.conf
 # Remove any platformid currently there
 sed -i -e 's, ignition.platform.id=[a-zA-Z0-9]*,,g' "${tmpd}"/bls.conf
-sed -i -e 's,^\(options .*\),\1 ignition.platform.id='"${platformid}"',' "${tmpd}"/bls.conf
+sed -i -e '/^options / s,$, ignition.platform.id='"${platformid}"',' "${tmpd}"/bls.conf
+if [ -n "$extra_kargs" ]; then
+    sed -i -e '/^options / s@$@ '"${extra_kargs}"'@' "${tmpd}"/bls.conf
+fi
 coreos_gf upload "${tmpd}"/bls.conf "${blscfg_path}"
 
+# Inject grub.cfg commands
+if [ -n "$extra_grub_cmds" ] && [ "$(coreos_gf exists /boot/grub2/grub.cfg)" = "true" ]; then
+    coreos_gf download /boot/grub2/grub.cfg "${tmpd}"/grub.cfg
+    sed -i -e 's@^# CONSOLE-SETTINGS$@'"${extra_grub_cmds}"'@' "${tmpd}"/grub.cfg
+    coreos_gf upload "${tmpd}"/grub.cfg /boot/grub2/grub.cfg
+fi
+
```

Comment:
does ppc64le use grub? just wondering if we need to conditionalize this?

Reply:
ppc64le doesn't use grub but petitboot reads the `grub.cfg`.  s390x doesn't use grub, but presumably won't define any `grub_commands`.  The only reason this block is conditionalized is so we don't try to inject the legacy grub commands on s390x.